contracts/tonomy/src/tonomy.cpp

```diff
@@ -497,4 +497,63 @@ namespace tonomysystem
           .send();
    }
 
+   void tonomy::staketokens(name account_name, asset quantity) {
+      require_auth(account_name);
+      eosio::check(quantity.amount > 0, "Stake quantity must be greater than zero.");
+      staking_allocations _staking(get_self(), get_self().value);
```

Comment:
better to use `account_name.value` for the scope here like we do https://github.com/Tonomy-Foundation/Tonomy-Contracts/blob/master/contracts/vesting.tmy/src/vesting.tmy.cpp#L38 which makes it easier to fetch the users total balance and iterate through all their allocations